python4j/python4j-numpy/src/main/java/org/nd4j/python4j/NumpyArray.java

```diff
@@ -282,31 +282,37 @@ public PythonObject toPython(INDArray indArray) {
         // PyArray_Type() call causes jvm crash in linux cpu if GIL is acquired by non main thread.
         // Using Interpreter for now:
 
-//        try(PythonContextManager.Context context = new PythonContextManager.Context("__np_array_converter")){
-//            log.info("Stringing exec...");
-//            String code = "import ctypes\nimport numpy as np\n" +
-//                    "cArr = (ctypes." + ctype + "*" + indArray.length() + ")"+
-//                    ".from_address(" + indArray.data().pointer().address() + ")\n"+
-//                    "npArr = np.frombuffer(cArr, dtype=" + ((numpyType == NPY_HALF) ? "'half'" : "ctypes." + ctype)+
-//                    ").reshape(" + Arrays.toString(indArray.shape()) + ")";
-//            PythonExecutioner.exec(code);
-//            log.info("exec done.");
-//            PythonObject ret = PythonExecutioner.getVariable("npArr");
-//            Py_IncRef(ret.getNativePythonObject());
-//            return ret;
-//
-//        }
-        log.info("NUMPY: PyArray_Type()");
-        PyTypeObject pyTypeObject = PyArray_Type();
-
-
-        log.info("NUMPY: PyArray_New()");
-        PyObject npArr = PyArray_New(pyTypeObject, shape.length, new SizeTPointer(shape),
-                numpyType, null,
-                inputArray.data().addressPointer(),
-                0, NPY_ARRAY_CARRAY, null);
-        log.info("Done.");
-        return new PythonObject(npArr);
+        //likely embedded in python, always use this method instead
+        if(!PythonConstants.releaseGilAutomatically() || PythonConstants.createNpyViaPython()) {
+            try(PythonContextManager.Context context = new PythonContextManager.Context("__np_array_converter")){
+                log.info("Stringing exec...");
+                String code = "import ctypes\nimport numpy as np\n" +
+                        "cArr = (ctypes." + ctype + "*" + indArray.length() + ")"+
+                        ".from_address(" + indArray.data().pointer().address() + ")\n"+
+                        "npArr = np.frombuffer(cArr, dtype=" + ((numpyType == NPY_HALF) ? "'half'" : "ctypes." + ctype)+
+                        ").reshape(" + Arrays.toString(indArray.shape()) + ")";
+                PythonExecutioner.exec(code);
+                log.info("exec done.");
```

Comment:
Does it make sense to keep this at info log level? or should this maybe move to the debug log level, as you did with the logging in the other branch?

Reply:
Fixed log levels.